services/slam/fake/data_loader.go

```diff
@@ -0,0 +1,145 @@
+package fake
+
+import (
+	"encoding/json"
+	"fmt"
+	"image"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+	"go.viam.com/utils"
+	"go.viam.com/utils/artifact"
+
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/spatialmath"
+	rdkutils "go.viam.com/rdk/utils"
+	"go.viam.com/rdk/vision"
+)
+
+type pose struct {
+	X float64 `json:"x"`
+	Y float64 `json:"y"`
+	Z float64 `json:"z"`
+}
+
+type poseInFrame struct {
+	Pose pose `json:"pose"`
+}
+
+type quat struct {
+	Imag float64 `json:"imag"`
+	Jmag float64 `json:"jmag"`
+	Kmag float64 `json:"kmag"`
+	Real float64 `json:"real"`
+}
+
+type extra struct {
+	Quat quat `json:"quat"`
+}
+
+type position struct {
+	Pose  poseInFrame `json:"pose"`
+	Extra extra       `json:"extra"`
+}
+
+// type positionNew struct {
+// 	Pose               pose   `json:"pose"`
+// 	ComponentReference string `json:"component_reference"`
+// 	Extra              extra  `json:"extra"`
+// }
+
+const (
+	internalStateTemplate = "%s/internal_state/internal_state_%d.pbstream"
+	maxDataCount          = 16
```

Comment:
[minor] isn't maxDataCount 10?

Reply:
Nope, there are 0-15 data files in the viam office dataset https://github.com/viamrobotics/rdk/pull/1897/files#diff-a941a642d16083a758932ab3b9201c6493dc650a0430b752a0ceac71d88e0e53R50656

Reply:
Ahh I got fooled by the ordering! Great!